recipes/cpr/all/conanfile.py

```diff
@@ -120,42 +154,88 @@ def _configure_cmake(self):
             self._cmake.definitions[self._get_cmake_option("CPR_BUILD_TESTS")] = False
             self._cmake.definitions[self._get_cmake_option("CPR_GENERATE_COVERAGE")] = False
             self._cmake.definitions[self._get_cmake_option("CPR_USE_SYSTEM_GTEST")] = False
-            if self._supports_openssl:
-                self._cmake.definitions["CMAKE_USE_OPENSSL"] = self.options.get_safe("with_openssl", False)
-            if self._supports_winssl: # The CMake options changed
-                # https://github.com/whoshuu/cpr/commit/18e1fc5c3fc0ffc07695f1d78897fb69e7474ea9#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR39-R40
-                self._cmake.definitions[self._get_cmake_option("CPR_FORCE_OPENSSL_BACKEND")] = self.options.get_safe("with_openssl", False)
-                self._cmake.definitions[self._get_cmake_option("CPR_FORCE_WINSSL_BACKEND")] = self.options.get_safe("with_winssl", False)
-
-            supports_any_ssl = self.options.get_safe("with_openssl", False) or self.options.get_safe("with_winssl", False)
-            if not self._uses_old_cmake_options and not supports_any_ssl:
+
+            ssl_value = self._get_ssl_library()
+            SSL_OPTIONS = {
+                "CPR_FORCE_DARWINSSL_BACKEND": ssl_value == "darwinssl",
+                "CPR_FORCE_OPENSSL_BACKEND": ssl_value == "openssl",
+                "CPR_FORCE_WINSSL_BACKEND": ssl_value == "winssl",
+                "CMAKE_USE_OPENSSL": ssl_value == "openssl"
+            }
+
+            for cmake_option, value in SSL_OPTIONS.items():
+                self._cmake.definitions[self._get_cmake_option(cmake_option)] = value
+
+            # If we are on a version where disabling SSL requires a cmake option, disable it
+            if not self._uses_old_cmake_options and self._get_ssl_library() == CprConan._NO_SSL:
                 self._cmake.definitions["CPR_ENABLE_SSL"] = False
+
             self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    # Check if the system supports the given ssl library
+    def _supports_ssl_library(self, library):
+        if library == CprConan._NO_SSL:
+            return True
+        elif library == CprConan._AUTO_SSL:
+            return self._can_auto_ssl
+
+        validators = {
+            "openssl": self._supports_openssl,
+            "darwinssl": self._supports_darwinssl,
+            "winssl": self._supports_winssl,
+            CprConan._AUTO_SSL: self._can_auto_ssl
+        }
+
+        if library not in validators:
+            # This should never happen, as the options are validated by conan.
+            raise ValueError("Unknown SSL library: {}".format(library))
+
```

Comment:
```suggestion
```

Let's just remove, because accessing `validators[library]` will raise more-or-less the same error and as you noted it's not gonna happen at all.